game/scripts/vscripts/abilities/oaa_bristleback.lua

```diff
@@ -0,0 +1,108 @@
+  local victim_angle = params.unit:GetAnglesAsVector().y
+  local origin_difference = params.unit:GetAbsOrigin() - params.attacker:GetAbsOrigin()
+  -- Get the radian of the origin difference between the attacker and Bristleback. We use this to figure out at what angle the attacker is at relative to Bristleback.
+  local origin_difference_radian = math.atan2(origin_difference.y, origin_difference.x)
+  -- Convert the radian to degrees.
+  origin_difference_radian = origin_difference_radian * 180
+  local attacker_angle = origin_difference_radian / math.pi
+  -- See the opening block comment for why I do this. Basically it's to turn negative angles into positive ones and make the math simpler.
+  attacker_angle = attacker_angle + 180.0
+  -- Finally, get the angle at which Bristleback is facing the attacker.
+  local result_angle = attacker_angle - victim_angle
+  result_angle = math.abs(result_angle)
+
+  -- Check for the side angle first. If the attack doesn't pass this check, we don't have to do back angle calculations.
+  if result_angle >= (180 - (ability:GetSpecialValueFor("side_angle") / 2)) and result_angle <= (180 + (ability:GetSpecialValueFor("side_angle") / 2)) then
+    -- Check for back angle. If this check doesn't pass, then do side angle "damage reduction".
+    if result_angle >= (180 - (ability:GetSpecialValueFor("back_angle") / 2)) and result_angle <= (180 + (ability:GetSpecialValueFor("back_angle") / 2)) then
+      -- Check if the reduced damage is lethal
+      if((params.unit:GetHealth() - (params.Damage * (1 - back_reduction_percentage))) >= 1) then
+        -- This is the actual "damage reduction".
+        params.unit:SetHealth((params.Damage * back_reduction_percentage) + params.unit:GetHealth())
+        -- Play the sound on Bristleback.
+        EmitSoundOn(params.sound, params.unit)
+        -- Create the back particle effect.
+        local back_damage_particle = ParticleManager:CreateParticle(params.back_particle, PATTACH_ABSORIGIN_FOLLOW, params.unit)
+        -- Set Control Point 1 for the back damage particle; this controls where it's positioned in the world. In this case, it should be positioned on Bristleback.
+        ParticleManager:SetParticleControlEnt(back_damage_particle, 1, params.unit, PATTACH_POINT_FOLLOW, "attach_hitloc", params.unit:GetAbsOrigin(), true)
+        -- Increase the Quill Spray damage counter based on how much damage was done *post-Bristleback mitigation*.
+        params.unit.quill_threshold_counter = params.unit.quill_threshold_counter + (params.Damage - (params.Damage * back_reduction_percentage))
+      end
+    else
+      -- Check if the reduced damage is lethal
+      if (params.unit:GetHealth() - (params.Damage * (1 - side_reduction_percentage))) >= 1 then
+        -- This is the actual "damage reduction".
+        params.unit:SetHealth((params.Damage * side_reduction_percentage) + params.unit:GetHealth())
```

Comment:
It is very akin to what Bristleback uses, though - it should stack additively with damage manipulators.